File: LLD/src/main/java/org/example/inheritance/child/Mentor.java
package org.example.inheritance.child;

import org.example.inheritance.base.User;

public class Mentor extends User {
    public String company;

    public Mentor(String name, String email, String company) {
        super(name, email);
        this.company = company;
        System.out.println("  [Mentor Constructor] Called for: " + this.name);
    }

    @Override
    public void solveProblem() {
        System.out.println(name + " (Mentor from " + company + ") is solving a System Design problem.");
    }
}

File: LLD NodeJs/p1_oops/c3_inheritance/child/Mentor.js
const User = require('../base/User');

/**
 * Mentor class extending User - demonstrates inheritance
 */
class Mentor extends User {
    company;

    constructor(name, email, company) {
        super(name, email);
        this.company = company;
        console.log("  [Mentor Constructor] Called for: " + this._name);
    }

    // Method override
    solveProblem() {
        console.log(this._name + " (Mentor from " + this.company + ") is solving a System Design problem.");
    }
}

module.exports = Mentor;
